doc/using/organizing.chapter.md

```diff
@@ -0,0 +1,96 @@
+# Organizing packages with callPackage {#chap-organizing}
+
+You can use [`stdenv.mkDerivation`](#sec-using-stdenv) to build packages, but it quickly becomes cumbersome when managing more than a small handful of packages. Nixpkgs provides a number of tools to help build and manage large sets of packages which depend on each other or on the Nixpkgs package set.
+
+## callPackage {#sec-callpackage}
+
+`pkgs.callPackage` is a tool for filling function arguments with `pkgs` attributes. For instance, if you have a file `foo.nix`:
+
+```nix
+{
+  stdenv,
+  fetchurl,
+  zlib,
+}:
+stdenv.mkDerivation {
+  name = "libfoo-1.2.3";
+  src = fetchurl {
+    url = "http://example.org/libfoo-1.2.3.tar.bz2";
+    hash = "sha256-tWxU/LANbQE32my+9AXyt3nCT7NBVfJ45CX757EMT3Q=";
+  };
+  nativeBuildInputs = [zlib];
+}
+```
+
+Then `pkgs.callPackage ./foo.nix { }` is equivalent to `import ./foo.nix { inherit (pkgs) stdenv fetchurl zlib; }`. The second argument to `callPackage` can be used to override arguments or to specify extra arguments that aren't in Nixpkgs.
```

Comment:
Not entirely true. `callPackage` also adds `.override` which can be used to override these arguments.